### DISCUSSION TITLE
deepmd-kit/lib/deepmd/libop_abi.so: undefined symbol: _ZN10tensorflow8internal21CheckOpMessageBuilder9NewStringB5cxx11Ev

### DISCUSSION THREAD

**Discussion by yanqinz2** [deepmodeling/deepmd-kit#15] [Q&A]
Dear all,

I installed the deepmd-kit on a linux environment exactly according to the manual. During the installation, there was no error generated. However, when I attempted to run the example "water.json", it generated a run-time error. The full error information is provided below:

$deepmd_root/bin/dp_train water.json
Traceback (most recent call last):
  File "/projects/zhyang/yanqinz2/packages/deepmd-kit/bin/dp_train", line 16, in <module>
    from deepmd.Model import NNPModel
  File "/projects/zhyang/yanqinz2/packages/deepmd-kit/bin/../lib/deepmd/Model.py", line 21, in <module>
    op_module = tf.load_op_library(module_path + "libop_abi.so")
  File "/projects/zhyang/yanqinz2/packages/tensorflow/tensorflow_venv/lib/python3.6/site-packages/tensorflow/python/framework/load_library.py", line 56, in load_op_library
    lib_handle = py_tf.TF_LoadLibrary(library_filename)
tensorflow.python.framework.errors_impl.NotFoundError: /projects/zhyang/yanqinz2/packages/deepmd-kit/lib/deepmd/libop_abi.so: undefined symbol: _ZN10tensorflow8internal21CheckOpMessageBuilder9NewStringB5cxx11Ev

I did activate the virtual environment before ran it. I have no idea what's wrong with it. Could you please give me some suggestion? I highly appreciate!

Best,
Yanqin

**Comment by amcadmus** [comment]
https://github.com/deepmodeling/deepmd-kit#training-tensorflow-abi-binary-cannot-be-found-when-doing-training

**Comment by yanqinz2** [comment]
Sorry for my missing the TroubleShooting part. That exactly solved my problem! Thank you very much!
